nservicebus/upgrades/5to6-moving-away-from-ibus.md

```diff
@@ -0,0 +1,95 @@
+
+If the endpoint is hosted using NServiceBus.Host, use the [IWantToRunWhenEndpointStartsAndStops interface](/nservicebus/upgrades/host-6to7.md) as outlined in the NServiceBus.Host documentation.
+
+## Sending messages inside message handlers
+
+The public `IBus` injection property in message handlers can now be safely deleted.
+
+The message handlers now includes an additional `IMessageHandlerContext` parameter, which provides the methods that used to be called from `IBus`. Use the `IMessageHandlerContext` to send and publish messages from within the message handler. 
+
+snippet:5to6-bus-send-publish
+
+
+## Accessing the CurrentMessageContext
+
+Previously it was possible to access message parameters such `MessageId`, `ReplyToAddress` and the message headers via the `CurrentMessageContext` property on the `IBus` interface. These message properties are now available directly via the message handler context parameter.
+
+snippet: 5to6-messagecontext
+
+
+## Dependency Injection
+
+In previous versions of NServiceBus the `IBus` interface was automatically registered in the IoC container. In Version 6, the new context aware interfaces namely the `IEndpointInstance`, `IMessageSession` and `IMessageHandlerContext` will not be automatically registered in the IoC container. 
+
+The `IEndpointInstance` interface can be injected into the container when self hosting. When using NServiceBus.Host, the `IMessageSession` interface can be injected into the container.
+
+### When injecting IBus in Custom Components
+
+When containers are used for dependency injection, if the `IBus` was previously being used in the injected custom classes, there are two options depending on the usage of the custom component and where it was invoked.
+
+#### Using the Custom Components outside of message handlers
+
+If the custom component was sending messages using the injected `IBus`from outside of message handlers, for example, in a MVC Controller class, then it is safe to use register either the `IEndpointInstance` when self hosting or register the `IMessageSession` when using the NServiceBus.Host in the container. This interface can then be used to send messages. 
+
+
+### Using the Custom Components inside message handlers
+
+However if the custom component is accessed from within message handlers in order to either send or publish events, then it is not safe to use the injected `IEndpointInstance` or `IMessageSession`. Instead the `IMessageHandlerContext` parameter should be passed to the custom component instead to either send or publish messages.  
+
+It is important to use the appropriate `IMessageHandlerContext` parameter from the message handler to send or publish messages.
+
+If the `IEndpointInstance` interface is used inside a message handler to send or publish messages, then:
```

Comment:
I'm with @timbussmann can we make this stronger?

"IEndpointInstance/MessageSession is not safe to use from message handlers. Some example of issues are..."

Also if we dial back on guidance on how to stick the above things into DI the users would have a much harder time getting those interfaces into the handlers? (statics, rebind in DI etc) => so must users would have troubles pointing the gun at their foot? (a good thing)